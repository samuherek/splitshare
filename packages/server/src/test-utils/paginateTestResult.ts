import { PaginationConnection } from '../types';

export default function paginateTestResult(
  resArray: any[]
): PaginationConnection<any> {
  return {
    edges: resArray.map((r) => ({ node: r, cursor: 'nodeCursor' })),
    pageInfo: {
      hasNextPage: false,
      endCursor: 'endCursor',
      itemsCount: 0,
    },
  };
}
