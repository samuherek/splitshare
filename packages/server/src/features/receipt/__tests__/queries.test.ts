import gql from 'graphql-tag';
import paginateTestResult from '../../../test-utils/paginateTestResult';
import createTestServer from '../../../test-utils/server';

const RECEIPTS = gql`
  query Receipts($billId: ID!) {
    receipts(billId: $billId) {
      edges {
        node {
          id
          comment
          category
          title
          total
          currency
          paidAt
          createdAt
          updatedAt
          isSettlement
        }
        cursor
      }
      pageInfo {
        hasNextPage
        endCursor
      }
    }
  }
`;

describe('receipt queries', () => {
  test('receipts: query should return an arry of receipts with appropriate fields', async () => {
    const { query } = await createTestServer({
      models: {
        Receipt: {
          getBillReceipts: jest.fn(() =>
            paginateTestResult([
              {
                id: 1,
                comment: null,
                category: null,
                title: 'title',
                total: 23.0,
                currency: 'EUR',
                paidAt: new Date('2000-01-01'),
                createdAt: new Date('2000-01-01'),
                updatedAt: new Date('2000-01-01'),
                isSettlement: false,
              },
            ])
          ),
        },
      },
    });

    const res = await query({ query: RECEIPTS, variables: { billId: 1 } });

    expect(res).toMatchSnapshot();
  });
});
