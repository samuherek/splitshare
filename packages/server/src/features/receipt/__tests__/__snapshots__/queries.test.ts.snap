// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receipt queries receipts: query should return an arry of receipts with appropriate fields 1`] = `
Object {
  "data": Object {
    "receipts": Object {
      "edges": Array [
        Object {
          "cursor": "nodeCursor",
          "node": Object {
            "category": null,
            "comment": null,
            "createdAt": "2000-01-01",
            "currency": "EUR",
            "id": "1",
            "isSettlement": false,
            "paidAt": "2000-01-01",
            "title": "title",
            "total": 23,
            "updatedAt": "2000-01-01T00:00:00.000Z",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "endCursor",
        "hasNextPage": false,
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
